### PR TITLE
Add ADR 0015 on managing access tokens in test

### PR DESCRIPTION
**Pull request by KPobeeNorris**
This continues the series of documenting decisions we made some time ago.

### CODE REVIEW

File: decisions/0015-manage-short-lived-access-tokens-in-tests.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+VCR allows you to filter out sensitive information, but this does not cover any data returned in the body.  We get a short-lived access token returned in the body as part of the OAuth flow and need to find a way of ensuring that these are not exposed and used to get access to sensitive data.  Whilst the tests are run against T3, which is an anonymised dataset, we don't want to risk the possibility of someone being able to engineer a way of undoing that and seeing real data.
+
+The main options:
+
+- making the github repository private
+- git-crypting the VCR cassettes folder
+- adding the VCR cassettes folder to .gitignore
+
+Making the github repository private, even for a very short period of time, as we are committed to coding in the open and it is very difficult to open up a repository once it has been made private.
```
Comment (rossjones) [inline decisions/0015-manage-short-lived-access-tokens-in-tests.md]: Making the github repository private contradicts our commitment to coding in the open, and eventually opening a previously closed repository is often risky and hard to prioritise.

**Review by rossjones — APPROVED** [review]

File: decisions/0015-manage-short-lived-access-tokens-in-tests.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+Date: 2019-01-16
+
+## Status
+
+Accepted
+
+## Context
+
+We have implemented the VCR gem as part of our testing suite; this gem allows you to "record your test suite's HTTP interactions and replay them during future test runs for fast, deterministic, accurate tests."
```
Comment (jennyd) [inline decisions/0015-manage-short-lived-access-tokens-in-tests.md]: I think we need to be explicit here that we've been committing the cassettes into the repo - without that context some of the options below don't make sense.

**Review by TobyRet — APPROVED** [review]

File: decisions/0015-manage-short-lived-access-tokens-in-tests.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+VCR allows you to filter out sensitive information, but this does not cover any data returned in the body.  We get a short-lived access token returned in the body as part of the OAuth flow and need to find a way of ensuring that these are not exposed and used to get access to sensitive data.  Whilst the tests are run against T3, which is an anonymised dataset, we don't want to risk the possibility of someone being able to engineer a way of undoing that and seeing real data.
+
+The main options:
+
+- making the github repository private
+- git-crypting the VCR cassettes folder
+- adding the VCR cassettes folder to .gitignore
```
Comment (jennyd) [inline decisions/0015-manage-short-lived-access-tokens-in-tests.md:19]: Also: not using VCR and instead using other ways of mocking external calls in tests

**Comment by rossjones** [comment]
Also just noticed, README (index) needs updating

File: decisions/0015-manage-short-lived-access-tokens-in-tests.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+## Decision
+
+We will add the VCR cassettes folder to gitignore.
+
+We will change the VCR config to allow a recording mode to be added i.e. "all" -> re-record all cassettes, "new_episodes" -> append the cassette(s) with any new calls made, "once" -> just record them once and reuse.
+
+## Consequences
+
+We remove the risk of short-lived access tokens being exposed publicly, we can continue to use VCR in local development to keep our test speeds as low as possible, and we will also continue to use VCR during our CI/CD workflow in CircleCi.
```
Comment (jennyd) [inline decisions/0015-manage-short-lived-access-tokens-in-tests.md:38]: Also: our tests will be making external network calls, which may increase build times and flakiness. We are likely to need to address these concerns later, but this decision is a pragmatic approach to help us move forward now.

Comment (jennyd) [reply decisions/0015-manage-short-lived-access-tokens-in-tests.md:38]: Also, related to ^, our CircleCI project envvars need to include credentials for authenticating against T3.

**Review by jennyd — APPROVED** [review]
👍 🎉 🌳

**Pull request merged**